### PR TITLE
twisted.logger, a whole bunch of PEP8, fixed the skipped test

### PR DESCRIPTION
**Pull request by anthonyalmarza**

### CODE REVIEW

**Comment by jMyles** [comment]
Whoa buddy!  I need to go get more coffee before I review this one.  :-)

File: hendrix/contrib/async/crosstown_traffic.py [hunk 1]
```diff
@@ -40,7 +39,11 @@ def __call__(self, crosstown_task):
         self.response = get_response_for_thread()
 
         if not self.no_go:
-            logger.info("Adding '%s' to crosstown_traffic for %s" % (crosstown_task.__name__, self.response))
+            self.log.info(
+                "Adding '{task!r}' to crosstown_traffic for "
```
Comment (jMyles) [inline hendrix/contrib/async/crosstown_traffic.py]: Nice - very elegant.

File: hendrix/deploy/base.py [hunk 2]
```diff
@@ -35,17 +35,21 @@ def __init__(self, action='start', options={}, reactor=reactor):
         # because running the management command overrides self.options['wsgi']
         if self.options['wsgi']:
             if hasattr(self.options['wsgi'], '__call__'):
-                # If it has a __call__, we assume that it is the application object itself.
+                # If it has a __call__, we assume that it is the application
+                # object itself.
                 self.application = self.options['wsgi']
                 try:
-                    self.options['wsgi_app_name'] = "%s.%s" % (self.application.__module__, self.application.__name__)
+                    self.options['wsgi'] = "%s.%s" % (
```
Comment (jMyles) [inline hendrix/deploy/base.py]: Can you help me understand this change?  Are we still allowing them to either pass an application object or a dot path?

File: hendrix/resources.py [hunk 3]
```diff
@@ -215,9 +215,6 @@ def DjangoStaticResource(path, rel_url='static'):
     rel_url = rel_url.strip('/')
     StaticFilesResource = MediaResource(path)
     StaticFilesResource.namespace = rel_url
-    chalk.green(
```
Comment (jMyles) [inline hendrix/resources.py]: Why eliminate this emission?  I found it very useful.

File: hendrix/test/resources.py [hunk 4]
```diff
@@ -17,82 +17,74 @@ class TestNameSpace(object):
...
 nameSpace = TestNameSpace()
 
 
 def application(environ, start_response):
-    start_response('200 OK', [('Content-type','text/plain')])
-    
+    start_response('200 OK', [('Content-type', 'text/plain')])
+
     if 'test_crosstown_traffic' in environ['QUERY_STRING'] or environ['PATH_INFO'] == '/r/1':
-        logger.info("Starting first cycle.")
```
Comment (jMyles) [inline hendrix/test/resources.py]: Hey Ant, why no logging for this test?  It's a complex back-and-forth test with the queues, so it's damn near impossible to tell why it fails without these emissions.

File: setup.py [hunk 5]
```diff
@@ -68,6 +68,9 @@ def mkdir_p(path):
     data_files=[
         (share_path, ['hendrix/utils/templates/init.d.j2', ]),
     ],
-    install_requires=readlines('requirements'),
-    extras_require={'ssl': ['pyopenssl', ]}
+    install_requires=readlines('requirements.txt'),
+    extras_require={
+        'ssl': ['pyopenssl', ],
+        'dev': ['django', 'flask', 'mock', 'pyopenssl', 'ipdb']
```
Comment (jMyles) [inline setup.py]: Ahhh, finally.  Nice.  Thanks man.

File: hendrix/deploy/base.py (hunk 2 above)
Comment (anthonyalmarza) [inline hendrix/deploy/base.py]: The point of this change was to accommodate whatever django is doing in their management command magic whilst not adding a new option. This way the user interface doesn't have to change and the skip on the test can be removed. `options` was only ever supposed to have user inputs/defaults entered from the terminal - at least in my mind. Not sure why the Django peeps would put a callable in there.

File: hendrix/resources.py (hunk 3 above)
Comment (anthonyalmarza) [inline hendrix/resources.py]: The emission is too verbose on larger projects. We can do better. Perhaps by adding a final statement after the statics are loaded (pointing to a log of the complete statics listing)

**Comment by anthonyalmarza** [comment]
I don't get that `ImportError` issue on either my mac or linux machines. I did get a `funcsigs` error. But have put that in the extra dev requirements.

In both of my environments I have `pip install -e .[dev]` which may be the difference between us.

`python hendrix/runtests.py` works fine for me.

File: hendrix/test/resources.py (hunk 4 above)
Comment (anthonyalmarza) [inline hendrix/test/resources.py]: I'm not a big fan of having log statements being emitted to the terminal during tests. But I understand what you mean. I'll put in a `log.debug` statement and later we can add a log analysis utility so to enable to easily resolve/find related test issues.

File: hendrix/deploy/base.py (hunk 2 above)
Comment (jMyles) [inline hendrix/deploy/base.py]: I see.  That makes sense - it did seem a little unseemly to me to have two options.

To answer your last sentence: Django (or Pyramid, or Flask) users might use an actual application object if they're calling hendrix from within Python and not from the command line.

File: hendrix/resources.py (hunk 3 above)
Comment (jMyles) [inline hendrix/resources.py]: Roger.  Makes sense.

Maybe have an optionally separate log for statics altogether?

File: hendrix/test/resources.py (hunk 4 above)
Comment (jMyles) [inline hendrix/test/resources.py]: Sounds good.

**Pull request merged**